File: Python/LAKI1/H5/month.py
# TIEP1.1 Lausekielinen ohjelmointi I
# Harjoitus 5, tehtävä 7
# Eveliina Toivanen (toivanen.eveliinarai@gmail.com)

# Ohjelma tulostaa käyttäjän antamaa numeroa vastaavan kuukauden sanakirjasta.


def lausu_kuukauden_nimi(numero):
    """"
    Funktio printtaa parametriarvonaan saadun numeron arvoparin sanakirjasta.
    """
    kuukaudet = {1: "January", 2: "February", 3: "March", 4: "April", 5: "May", 6: "June",
                 7: "July", 8: "August", 9: "September", 10: "October", 11: "November",
                 12: "December"}
    print("It is ", kuukaudet[numero], ".", sep="")


def main():
    """"
    Funktio tervehtii käyttäjää ja pyytää tältä numeroa, jonka funktio lähettää
    lausu_kuukauden_nimi -funktiolle.
    """
    print("Hello! I verbalise your input.")
    print("Please, enter a number:")
    kuukauden_numero = int(input())

    lausu_kuukauden_nimi(kuukauden_numero)


# Kutsutaan pääohjelmaa, jos ohjelmaa ajetaan.
if __name__ == "__main__":
    main()

File: Java/Laki2/H4/Month.java
/* TIEP5.1 Lausekielinen ohjelmointi II
 * Harjoitus 4, tehtävä 1
 * Eveliina Toivanen (toivanen.eveliinarai@gmail.com)
 *
 * Luetaan käyttäjältä järjestysluku ja tulostetaan sen mukainen kuukausi.
 * Tein tämän näin jo alunperin tällä tavalla.
 */
 
import java.util.Scanner;

public class Month {
    
    /* Metodissa tulostetaan käyttäjän antaman järjestysluvun mukainen kuukausi.
     * kuukaudenNumero: Käyttäjän antama kuukauden järjestysluku.
     */
    public static void lausuKuukaudenNimi(int kuukaudenNumero){
         
         String[] kuukaudet = {"January", "February", "March", "April", "May", "June", 
                               "July", "August", "September", "October", "November", "December"};
                               
         // Vähennetään annetusta numerosta yksi, jotta luku täsmää taulukon indeksin kanssa.
         System.out.println("It is " + kuukaudet[kuukaudenNumero - 1] + ".");
     }
    
    /* Pääohjelmassa tervehditään käyttäjää, pyydetään tältä syötteenä kuukauden järjestysluku
     * ja lähetetään syöte lausuKuukaudenNimi metodille.
     */
    public static void main(String[] args) {
        
        // Alustetaan syötteenlukija.
        Scanner lukija = new Scanner(System.in);
         
         System.out.println("Hello! I verbalise your input.");
         System.out.println("Please, enter a number:");
         int kuukaudenNumero = lukija.nextInt();
         
         lausuKuukaudenNimi(kuukaudenNumero);
    }
}